docs/content/customization/aws/ami.md

```diff
@@ -0,0 +1,86 @@
++++
+title = "AWS AMI ID and Format spec"
++++
+
+The AWS AMI customization allows the user to specify the AMI or AMI Lookup arguments for a AWS machine.
+The AMI customization can be applied to both control plane and nodepool machines.
+This customization will be available when the
+[provider-specific cluster configuration patch]({{< ref "..">}}) is included in the `ClusterClass`.
+
+## Example
+
+To specify the AMI ID or format for all control plane and nodepools, use the following configuration:
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  name: <NAME>
+spec:
+  topology:
+    variables:
+      - name: clusterConfig
+        value:
+          controlPlane:
+            aws:
+              ami:
+                id: "ami-controlplane"
+                lookupFormat: "my-cp-ami-{{.BaseOS}}-?{{.K8sVersion}}-*"
+                lookupOrg: "123456789"
+                lookupBaseOS: "ubuntu-20.04"
+      - name: workerConfig
+        value:
+          aws:
+            ami:
+                id: "ami-allWorkers"
+                lookupFormat: "my-default-workers-ami-{{.BaseOS}}-?{{.K8sVersion}}-*"
+                lookupOrg: "123456789"
+                lookupBaseOS: "ubuntu-20.04"
+```
+
+We can further customize individual MachineDeployments by using the overrides field with the following configuration:
+
+```yaml
+spec:
+  topology:
+    # ...
+    workers:
+      machineDeployments:
+        - class: default-worker
+          name: md-0
+          variables:
+            overrides:
+              - name: workerConfig
+                value:
+                   ami:
+                    id: "ami-customWorker"
+                    lookupFormat: "gpu-workers-ami-{{.BaseOS}}-?{{.K8sVersion}}-*"
+                    lookupOrg: "123456789"
+                    lookupBaseOS: "ubuntu-20.04"
```

Comment:
```suggestion
                    # set either ID or some combination of lookup values
                    id: "ami-customWorker"
                    lookupFormat: "gpu-workers-ami-{{.BaseOS}}-?{{.K8sVersion}}-*"
                    lookupOrg: "123456789"
                    lookupBaseOS: "ubuntu-20.04"
```

Reply:
We do not validate or restrict user to provide either ID or lookup values. user can provide both, but ami id takes precedence. I have added that to the variable description. user is allowed to specify both. 
I think we want to keep them mutually exclusive then that logic should go in the client. WDYT?

Reply:
What about having `id` and `lookup` (object with properties) as keys here?